core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -350,6 +346,37 @@ private static Stream<DeleteFile> limitBySequenceNumber(
     return Arrays.stream(files, start, files.length);
   }
 
+  private static DeleteFileGroup index(
```

Comment:
These `index` methods only exist because the old constructor is package private and is used in tests. I had to keep that for compatibility.